WORKSPACE

```diff
@@ -611,18 +611,83 @@ load("@rules_jvm_external//:defs.bzl", "maven_install")
 
 maven_install(
     artifacts = [
-       "com.google.guava:guava:31.1-jre",
-       "com.google.guava:guava-testlib:31.1-jre",
-       "com.google.guava:failureaccess:1.0.1",
-       "com.google.errorprone:error_prone_annotations:2.16",
-       "com.google.errorprone:error_prone_type_annotations:2.16",
-       "com.google.code.findbugs:jsr305:3.0.2",
-       "com.google.j2objc:j2objc-annotations:1.3",
-       "com.github.stephenc.jcip:jcip-annotations:1.0-1",
-       "org.checkerframework:checker-qual:3.12.0",
+        "com.github.ben-manes.caffeine:caffeine:3.0.5",
+        "com.github.kevinstern:software-and-algorithms:1.0",
+        "com.github.stephenc.jcip:jcip-annotations:1.0-1",
+        "com.google.api-client:google-api-client-gson:1.35.2",
+        "com.google.api-client:google-api-client:1.35.2",
+        "com.google.auth:google-auth-library-credentials:1.6.0",
+        "com.google.auth:google-auth-library-oauth2-http:1.6.0",
+        "com.google.auto.service:auto-service-annotations:1.0.1",
+        "com.google.auto.service:auto-service:1.0",
+        "com.google.auto.value:auto-value-annotations:1.9",
+        "com.google.auto.value:auto-value:1.8.2",
+        "com.google.auto:auto-common:1.2.1",
+        "com.google.code.findbugs:jsr305:3.0.2",
+        "com.google.code.gson:gson:2.9.0",
+        "com.google.code.java-allocation-instrumenter:java-allocation-instrumenter:3.3.0",
+        "com.google.errorprone:error_prone_annotation:2.16",
+        "com.google.errorprone:error_prone_annotations:2.16",
+        "com.google.errorprone:error_prone_check_api:2.16",
+        "com.google.errorprone:error_prone_core:2.16",
+        "com.google.errorprone:error_prone_type_annotations:2.16",
+        "com.google.flogger:flogger-system-backend:0.5.1",
+        "com.google.flogger:flogger:0.5.1",
+        "com.google.flogger:google-extensions:0.5.1",
+        "com.google.guava:failureaccess:1.0.1",
+        "com.google.guava:guava-testlib:31.1-jre",
+        "com.google.guava:guava:31.1-jre",
+        "com.google.http-client:google-http-client-gson:1.41.4",
+        "com.google.http-client:google-http-client:1.41.4",
+        "com.google.j2objc:j2objc-annotations:1.3",
+        "com.squareup:javapoet:1.8.0",
```

Comment:
Suggestion: use the `maven.artifact` extension to specify `neverlink` on artifacts like javapoet: https://github.com/bazelbuild/rules_jvm_external#compile-only-dependencies and `testonly` on test artifacts: https://github.com/bazelbuild/rules_jvm_external#test-only-dependencies

Reply:
Done for guava-testlib and javapoet, but it's a bit unclear to me which libraries should be `neverlink`.

Reply:
Hmm, it seems javapoet cannot be neverlink, I'm getting:
```
Caused by: java.lang.NoClassDefFoundError: com/squareup/javapoet/TypeSpec
	at com.google.devtools.build.lib.skyframe.serialization.autocodec.SerializationConstantProcessor.buildRegisteredSingletonClass(SerializationConstantProcessor.java:81)
	at com.google.devtools.build.lib.skyframe.serialization.autocodec.SerializationConstantProcessor.processInternal(SerializationConstantProcessor.java:66)
	at com.google.devtools.build.lib.skyframe.serialization.autocodec.SerializationConstantProcessor.process(SerializationConstantProcessor.java:54)
	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:985)
	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:901)
	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1227)
	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1340)
	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1258)
	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:936)
	at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.lambda$doCall$0(JavacTaskImpl.java:104)
	at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.handleExceptions(JavacTaskImpl.java:147)
```